src/persist-types/src/lib.rs

```diff
@@ -110,6 +110,13 @@ pub trait Codec: Default + Sized + PartialEq + 'static {
         Ok(())
     }
 
+    /// Checks that the given value matches the provided schema.
+    ///
+    /// A no-op default implementation is provided for convenience.
+    fn validate(_val: &Self, _schema: &Self::Schema) -> Result<(), String> {
+        Ok(())
+    }
```

Comment:
I thought of this, but decided against it because the compaction case would have to decode and re-encode, vs decode and validate. It's CI-only and that's probably not a huge perf difference, but still feels bad to slow down CI more than necessary. Plus a bonus is that this sticks around even if we delete `Codec::encode` and `Codec::decode` tomorrow.